src/api/v1/serializers/pizzas-serializer.js

```diff
@@ -98,9 +105,13 @@ const serializePizza = (rawPizza, query) => {
   let options = serializerOptions(serializerArgs);
   if ('dough' in rawPizza) {
     options = addCompoundRelationship(options, 'dough', doughResourceKeys);
+  } else {
+    options = addCompoundRelationship(options, 'dough', doughResourceKeys, true);
   }
   if ('ingredients' in rawPizza) {
     options = addCompoundRelationship(options, 'ingredients', ingredientResourceKeys);
+  } else {
+    options = addCompoundRelationship(options, 'ingredients', ingredientResourceKeys, true);
```

Comment:
The logic of line 106-110 is exactly the same as line 111-114. I guess it should be possible to combine together. The total line amount might not be improved a lot at this moment, but if we have more resources that can be included in the future or need to add more logic check here, it will be easier for you to extend/maintain.